src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,115 @@
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(InvalidDataException.class, () -> registrationService.register(null));
+        String actual = assertThrows(InvalidDataException.class, () ->
+                registrationService.register(null)).getMessage();
+        assertEquals("User can't be null", actual);
+    }
+
+    @Test
+    void register_nullLogin_NotOk() {
+        user.setLogin(null);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+        String actual = assertThrows(InvalidDataException.class, () ->
+                registrationService.register(user)).getMessage();
+        assertEquals("Login can't be null", actual);
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+        String actual = assertThrows(InvalidDataException.class, () ->
+                registrationService.register(user)).getMessage();
+        assertEquals("Age can't be null", actual);
+
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(null);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+        String actual = assertThrows(InvalidDataException.class, () ->
+                registrationService.register(user)).getMessage();
+        assertEquals("Password can't be null", actual);
+
+    }
+
+    @Test
+    void register_ageLess18_notOk() {
+        user.setAge(INVALID_AGE);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+        String actual = assertThrows(InvalidDataException.class, () ->
+                registrationService.register(user)).getMessage();
+        assertEquals("Age can't be less than 18", actual);
+
+    }
+
+    @Test
+    void register_PasswordLengthLess6Chars_notOk() {
+        user.setPassword(INVALID_PASSWORD);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+        String actual = assertThrows(InvalidDataException.class, () ->
+                registrationService.register(user)).getMessage();
+        assertEquals("Password's length can't be less than 6 characters", actual);
+    }
+
+    @Test
+    void register_LoginLengthLess6Chars_notOk() {
+        user.setLogin(INVALID_LOGIN);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(user));
+        String actual = assertThrows(InvalidDataException.class, () ->
+                registrationService.register(user)).getMessage();
+        assertEquals("Login can't be  less than 6 characters", actual);
+    }
+
+    @Test
+    void register_ExistUser_notOk() {
+        Storage.people.add(user);
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user),
+                "Added the same user");
+    }
+

```

Comment:
string param in the `assertThrows()` is a custom message that we can add and receive when the test fails, needed to describe the reason why a test failed
We should also assert the exception message as I said earlier
```suggestion
        InvalidDataException ex = assertThrows(InvalidDataException.class,
                () -> registrationService.register(user),
                "we shouldn't allow to register a user with the same login as one of the existing users"); 
        // assert that ex.getMessage() is equal to some expected message
    }
```

Reply:
Ok) I understand!! Thank you))